src/main/java/taxiservice/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,238 @@
+        }
+
+        query = "DELETE FROM " + CARS_DRIVERS_TAB_NAME
+                + " WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update " + car, e);
+        }
+
+        query = "INSERT INTO " + CARS_DRIVERS_TAB_NAME
+                + " (car_id, diver_id) VALUES (?, ?)";
+        for (Driver driver : car.getDrivers()) {
+            try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement = connection
+                             .prepareStatement(query)) {
+                statement.setLong(1, car.getId());
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            } catch (SQLException e) {
+                throw new DataProcessingException("Can't update " + car, e);
+            }
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE " + CARS_TAB_NAME
+                + " SET deleted = true"
+                + " WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> result = new ArrayList<>();
+        String query = "SELECT * FROM " + CARS_DRIVERS_TAB_NAME + " cd"
+                + " JOIN " + CARS_TAB_NAME + " c"
+                + " ON cd.car_id = c.car_id"
+                + " JOIN " + MANUFACTURERS_TAB_NAME + " m"
+                + " ON c.manufacturer_id = m.manufacturer_id"
+                + " JOIN " + DRIVERS_TAB_NAME + " d"
+                + " ON cd.diver_id = d.driver_id"
+                + " WHERE cd.diver_id = ? AND c.deleted = false AND d.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                result.add(newCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all by driver", e);
+        }
+        if (result != null) {
+            for (Car car : result) {
+                car.setDrivers(getDrivers(car.getId()));
```

Comment:
```suggestion
                car.setDrivers(getDrivers(car.getId()));
```
put this method inside try/catch and use one connection please